pkg/driver/mysql.go

```diff
@@ -0,0 +1,119 @@
+package driver
+
+import (
+	"context"
+	"database/sql/driver"
+	"github.com/go-sql-driver/mysql"
+	"github.com/pkg/errors"
+	"strconv"
+)
+
+// MySQLDriver extends mysql.MySQLDriver with auto-SETting Galera cluster options.
+type MySQLDriver struct {
+	mysql.MySQLDriver
+}
+
+// Open implements the driver.Driver interface.
+func (md MySQLDriver) Open(name string) (driver.Conn, error) {
+	connector, err := md.OpenConnector(name)
+	if err != nil {
+		return nil, err
+	}
+
+	return connector.Connect(context.Background())
+}
+
+// OpenConnector implements the driver.DriverContext interface.
+func (md MySQLDriver) OpenConnector(name string) (driver.Connector, error) {
+	var wsrepSyncWait int64
+
+	if config, err := mysql.ParseDSN(name); err == nil {
+		if s, ok := config.Params["wsrep_sync_wait"]; ok {
+			if i, err := strconv.ParseInt(s, 10, 64); err == nil {
+				// MySQL single nodes don't know wsrep_sync_wait and fail with error 1193 "Unknown system variable".
+				// We have to SET it manually later and swallow error 1193 not to fail our connections.
+				wsrepSyncWait = i
+				delete(config.Params, "wsrep_sync_wait")
+				name = config.FormatDSN()
+			}
+		}
```

Comment:
It should actually be possible to completely avoid the DSN for connection to a database by chaining these functions:

- https://pkg.go.dev/github.com/go-sql-driver/mysql#NewConnector
- https://pkg.go.dev/database/sql#OpenDB
- https://pkg.go.dev/github.com/jmoiron/sqlx#NewDb

This should then allow simply passing the config option as a struct member instead.

(And maybe this could also allow us to avoid having to register the custom `"icingadb-mysql"` driver at all.)

Reply:
Why just MySQL? Our Pg driver has a NewConnector() as well.

And what's the problem in general? Describing a connection to a database is the DSN's _purpose_.